src/main/groovy/com/bmuschko/gradle/docker/utils/RegistryAuthLocator.groovy

```diff
@@ -0,0 +1,247 @@
+package com.bmuschko.gradle.docker.utils
+
+import com.fasterxml.jackson.databind.JsonNode
+import com.fasterxml.jackson.databind.ObjectMapper
+import com.github.dockerjava.api.model.AuthConfig
+import groovy.transform.CompileStatic
+import org.apache.commons.lang.SystemUtils
+import org.gradle.api.logging.Logger
+import org.gradle.api.logging.Logging
+
+/**
+ * Utility class to get credentials information from $DOCKER_CONFIG/.docker/config.json file
+ * Supports auth token, credentials store and credentials helpers
+ * Only linux OS is supported at the moment.
+ * Returns default auth object if called on windows
```

Comment:
Could you please reference the code repository you got this from as a comment to give credit?

Reply:
Done